### PR TITLE
FI-479 Multiple Or Tests

### PR DESCRIPTION
**Pull request by czh-orz**
This adds tests for multipleOr and  refactors resolve_element_from_path so we don't need can_resolve_path anymore. Tests for multipleAnd is only SHOULD's and can be added later. My strategy for multipleOr's was to look through the returned resources and look for two different values for the search param. If the second value was not found, then skip the test. I'm not sure if this is the best way so I'm open to suggestions. 

**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-479
- [X] Internal ticket links to this PR
- [X] Internal ticket is properly labeled (Community/Program)
- [X] Internal ticket has a justification for its Community/Program label
- [X] Code diff has been reviewed for extraneous/missing code
- [ ] Tests are included and test edge cases
- [X] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by Jammjammjamm** [comment]
Can you rebase this on the most recent version of development?

**Comment by czh-orz** [comment]
Rebased to development

File: generator/uscore/metadata_extractor.rb [hunk 1]
```diff
@@ -224,6 +224,10 @@ def add_search_param_descriptions(profile_definition, sequence)
             expectation = expectation_extension[index]['extension'].first['valueCode'] unless expectation_extension.nil?
             param_metadata[:comparators][comparator.to_sym] = expectation
           end
+          multiple_or_expectation = search_param_definition['_multipleOr']['extension'].first['valueCode']
+          multiple_and_expectation = search_param_definition['_multipleAnd']['extension'].first['valueCode']
```
Comment (Jammjammjamm) [inline generator/uscore/metadata_extractor.rb]: If this PR isn't doing anything with multiple and, then there shouldn't be any multiple and code.

File: lib/modules/uscore_v3.1.0/pediatric_bmi_for_age_sequence.rb [hunk 2]
```diff
@@ -208,6 +208,14 @@ def validate_resource_item(resource, property, value)
         reply = get_resource_by_params(versioned_resource_class('Observation'), search_params)
         validate_search_reply(versioned_resource_class('Observation'), reply, search_params)
         assert_response_ok(reply)
+
+        second_value = resolve_element_from_path(@observation_ary, 'status') { |el| get_value_for_search_param(el) != search_params[:status] }
```
Comment (Jammjammjamm) [inline lib/modules/uscore_v3.1.0/pediatric_bmi_for_age_sequence.rb]: I think these should be separate tests.

File: lib/modules/uscore_v3.1.0/pediatric_weight_for_height_sequence.rb [hunk 3]
```diff
@@ -16,26 +16,26 @@ def validate_resource_item(resource, property, value)
...
 
         when 'code'
-          value_found = can_resolve_path(resource, 'code.coding.code') { |value_in_resource| value_in_resource == value }
-          assert value_found, 'code on resource does not match code requested'
+          value_found = resolve_element_from_path(resource, 'code.coding.code') { |value_in_resource| value.split(',').include? value_in_resource }
+          assert value_found.present?, 'code on resource does not match code requested'
 
         when 'date'
-          value_found = can_resolve_path(resource, 'effectiveDateTime') do |date|
+          value_found = resolve_element_from_path(resource, 'effectiveDateTime') do |date|
```
Comment (radamson) [inline lib/modules/uscore_v3.1.0/pediatric_weight_for_height_sequence.rb]: I'm assuming rubocop changed this to a `do`, but it might be nice to have the generator do them all this way for a consistent look.  Thoughts?

File: generator/uscore/uscore_generator.rb [hunk 4]
```diff
@@ -377,6 +377,38 @@ def create_resource_profile_test(sequence)
...
 
+      def create_multiple_or_test(sequence)
+        test = {
+          tests_that: 'The server returns expected results when parameters use composite-or',
+          index: sequence[:tests].length + 1,
+          link: sequence[:profile],
+          test_code: ''
+        }
+
+        multiple_or_params = sequence[:search_param_descriptions]
```
Comment (Jammjammjamm) [inline generator/uscore/uscore_generator.rb]: This seems like it could be its own method.

File: generator/uscore/uscore_generator.rb [hunk 5]
```diff
@@ -377,6 +377,38 @@ def create_resource_profile_test(sequence)
...
+          .select { |_param, description| description[:multiple_or] == 'SHALL' }
+          .map { |param, _description| param.to_s }
+        return if multiple_or_params.blank?
```
Comment (Jammjammjamm) [inline generator/uscore/uscore_generator.rb]: This return isn't actually needed.

File: lib/modules/uscore_v3.1.0/us_core_careteam_sequence.rb [hunk 6]
```diff
@@ -236,9 +229,33 @@ def validate_resource_item(resource, property, value)
         @instance.save!
       end
 
-      test 'All references can be resolved' do
+      test 'The server returns expected results when parameters use composite-or' do
```
Comment (Jammjammjamm) [inline lib/modules/uscore_v3.1.0/us_core_careteam_sequence.rb]: Is it really just CareTeam and MedicationRequest that have multiple or?

File: generator/uscore/uscore_generator.rb (hunk 4 above)
Comment (czh-orz) [reply generator/uscore/uscore_generator.rb]: Moved to its own method

File: generator/uscore/uscore_generator.rb (hunk 5 above)
Comment (czh-orz) [reply generator/uscore/uscore_generator.rb]: Removed

File: lib/modules/uscore_v3.1.0/us_core_careteam_sequence.rb (hunk 6 above)
Comment (czh-orz) [reply lib/modules/uscore_v3.1.0/us_core_careteam_sequence.rb]: They're the only  ones with required searches

File: generator/uscore/uscore_generator.rb (hunk 4 above)
Comment (Jammjammjamm) [reply generator/uscore/uscore_generator.rb]: You should use that method here 😉

Comment (czh-orz) [reply generator/uscore/uscore_generator.rb]: fixed

File: lib/modules/uscore_v3.1.0/pediatric_weight_for_height_sequence.rb (hunk 3 above)
Comment (arscan) [reply lib/modules/uscore_v3.1.0/pediatric_weight_for_height_sequence.rb]: Fixed.
